clouddrift/select.py

```diff
@@ -50,30 +51,34 @@ def subset(ds: ak.Array, criteria: dict) -> ak.Array:
         A scalar for selecting a specific value.
         >>> subset(ds, {"drogue_status": True})  # extract segment of trajectory with drogue
     """
-
-    mask_traj = ak.ones_like(ds[ds.fields[0]], dtype="bool")
-    mask_obs = ak.ones_like(ds.obs[ds.obs.fields[0]], dtype="bool")
+    mask_traj = xr.DataArray(data=np.ones(ds.dims["traj"], dtype="bool"), dims=["traj"])
+    mask_obs = xr.DataArray(data=np.ones(ds.dims["obs"], dtype="bool"), dims=["obs"])
 
     for key in criteria.keys():
-        if key in ds.fields:
-            mask_traj = np.logical_and(mask_traj, mask_var(ds[key], criteria[key]))
-        elif key in ds.obs.fields:
-            mask_obs = np.logical_and(mask_obs, mask_var(ds.obs[key], criteria[key]))
+        if key in ds:
+            if ds[key].dims == ("traj",):
+                mask_traj = np.logical_and(mask_traj, mask_var(ds[key], criteria[key]))
+            elif ds[key].dims == ("obs",):
+                mask_obs = np.logical_and(mask_obs, mask_var(ds[key], criteria[key]))
         else:
             raise ValueError(f"Unknown variable '{key}'.")
```

Comment:
This errors out if there is a variable that does not have `traj` or `dims`. Should we warn and let it run through instead? It seems reasonable to me that a user could have some arbitrary data here using other dims.

Reply:
Never mind, I made a blunder, all good.